File: src/models/notebook-delay.ts
export type NotebookDelay = Readonly<{
  id: string;
  date: Date;
  minutes: number;
  justified: boolean;
  justification?: string;
  shouldParentsJustify: boolean;
  administrativelyFixed: boolean;
  isReasonUnknown: boolean;
  reason?: string;
}>;

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/NotebookDelay.kt
package ink.literate.pawnote.models

import kotlinx.datetime.LocalDateTime

data class NotebookDelay(
    val id: String,
    val date: LocalDateTime,
    val minutes: Int,
    val justified: Boolean,
    val justification: String? = null,
    val shouldParentsJustify: Boolean,
    val administrativelyFixed: Boolean,
    val isReasonUnknown: Boolean,
    val reason: String? = null
)